### PR TITLE
Use CardTable::card_size_in_words rather than hard coded constant

### PR DESCRIPTION
**Pull request by earthling-amzn**
Calculation assumed 64 words per card, which does not hold on 32 bit word platforms. The number of words per card also depends on `GCCardSizeInBytes` command line parameter.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Author) ⚠️ Review applies to [9ebacb5d](https://git.openjdk.org/shenandoah/pull/186/files/9ebacb5defdead91b4c62ec116a3952508e52249)
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah pull/186/head:pull/186` \
`$ git checkout pull/186`

Update a local copy of the PR: \
`$ git checkout pull/186` \
`$ git pull https://git.openjdk.org/shenandoah pull/186/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 186`

View PR using the GUI difftool: \
`$ git pr show -t 186`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/186.diff">https://git.openjdk.org/shenandoah/pull/186.diff</a>

</details>

### CODE REVIEW

**Review by ysramakrishna — APPROVED** [review]

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp [hunk 1]
```diff
@@ -1024,16 +1024,17 @@ class ShenandoahRegionChunkIterator : public StackObj {
...
 
   static const size_t _clusters_in_smallest_chunk = 4;
-  static const size_t _assumed_words_in_card = 64;
 
   // smallest_chunk_size is 4 clusters (i.e. 128 KiB).  Note that there are 64 words per card and there are 64 cards per
   // cluster.  Each cluster spans 128 KiB.
   // This is computed from CardTable::card_size_in_words() *
   //      ShenandoahCardCluster<ShenandoahDirectCardMarkRememberedSet>::CardsPerCluster;
   // We can't perform this computation here, because of encapsulation and initialization constraints.  We paste
   // the magic number here, and assert that this number matches the intended computation in constructor.
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp:1033]: Is the portion of the comment beginning "We paste the maginc number ... etc." not obsolete now, and should there be deleted?

Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp:1033]: Yes - I'll clean that up.

**Review by kdnilsen — APPROVED** [review]
Thanks.

**Pull request closed without merging**